src/internet/radiobrowser/radiobrowserservice.h

```diff
@@ -0,0 +1,145 @@
+/* This file is part of Clementine.
+   Copyright 2021, Fabio Bas <ctrlaltca@gmail.com>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef INTERNET_RADIOBROWSER_RADIOBROWSERSERVICE_H_
+#define INTERNET_RADIOBROWSER_RADIOBROWSERSERVICE_H_
+
+#include <QJsonObject>
+
+#include "core/cachedlist.h"
+#include "internet/core/internetmodel.h"
+#include "internet/core/internetservice.h"
+
+class RadioBrowserUrlHandler;
+
+class QNetworkAccessManager;
+class QNetworkReply;
+class QMenu;
+
+class RadioBrowserService : public InternetService {
+  Q_OBJECT
+
+ public:
+  RadioBrowserService(Application* app, InternetModel* parent);
+  ~RadioBrowserService();
+
+  enum ItemType {
+    Type_Stream = 2000,
+  };
+
+  enum Type {
+    Type_Category = InternetModel::TypeCount,
+    Type_CategoryItem,
+    Type_Top100,
+  };
+
+  enum Role {
+    Role_ListUrl = InternetModel::RoleCount,
+    Role_ItemsUrl,
+  };
+
+  struct Stream {
+    QString name_;
+    QUrl url_;
+    QUrl favicon_;
+
+    Song ToSong(const QString& prefix) const;
+  };
+
+  struct Branch {
+    QString name;
+    QString listUrl;
+    QString itemsUrl;
+    Type type;
+  };
+
+  static QList<Branch> BranchList;
```

Comment:
I don't think these need to be public members, I think they can be in the `.cpp` in an anonymous namespace.

Reply:
`kServiceName` is accessed by `RadioBrowserSearchProvider` and required by `InternetModel::Service()`
`kSchemeName` is accessed by `RadioBrowserUrlHandler`
`kSettingsGroup` and `defaultServer` are accessed by `RadioBrowserSettingsPage`.
I tried to follow the way other services are implemented, I have to admin that the result is quite ugly, but probably the solution would be to refactor `InternetService` to make room for these values that are used by almost all of its subclasses.

Reply:
Yes, the `k*` constants are used elsewhere but `SearchUrl`, `BranchList`, `Branch`, etc. don't need to be exposed.